api/client/build.go

```diff
@@ -317,9 +317,17 @@ func getDockerfileRelPath(givenContextDir, givenDockerfile string) (absContextDi
 
 	// The context dir might be a symbolic link, so follow it to the actual
 	// target directory.
-	absContextDir, err = filepath.EvalSymlinks(absContextDir)
-	if err != nil {
-		return "", "", fmt.Errorf("unable to evaluate symlinks in context path: %v", err)
+	//
+	// FIXME. This is a hack which needs an issue against golang. On Windows,
+	// EvalSymlinks does not work on a UNC file path (ie one starting \\).
+	// All this means is that on Windows and on UNC paths, links will not
+	// be followed.
+	if runtime.GOOS != "windows" ||
+		(runtime.GOOS == "windows" && absContextDir[:2] != `\\`) {
+		absContextDir, err = filepath.EvalSymlinks(absContextDir)
+		if err != nil {
+			return "", "", fmt.Errorf("unable to evaluate symlinks in context path: %v", err)
+		}
```

Comment:
Can we factor the test out in a function? Also this can probably be simplified:

```
// FIXME. On Windows, EvalSymlinks does not work on a UNC file
// path (i.e. one starting with \\).
// All this means is that on Windows and on UNC paths, links will not
// be followed.
func isUNC(path string) bool {
    return runtime.GOOS == "windows" && strings.HasPrefix(path, `\\`)
}
```

And further down:

    if !isUNC(absContextDir) {
        [...]
    }